### PR TITLE
test stability of sort and sortOn

### PR DESCRIPTION
**Pull request by oisdk**
as described in #509

### CODE REVIEW

File: tests/seq-properties.hs [hunk 1]
```diff
@@ -560,6 +562,23 @@ prop_sort :: Seq OrdA -> Bool
 prop_sort xs =
     toList' (sort xs) ~= Data.List.sort (toList xs)
 
+data UnstableOrd = UnstableOrd
+    { ordKey :: OrdA
+    , _val :: A
```
Comment (treeowl) [inline tests/seq-properties.hs]: Maybe `_ignored`?

File: tests/seq-properties.hs [hunk 2]
```diff
@@ -560,6 +562,23 @@ prop_sort :: Seq OrdA -> Bool
...
+    } deriving (Eq,Show)
```
Comment (treeowl) [inline tests/seq-properties.hs]: No, definitely don't derive `Eq`!!! It must match the `Ord` instance! You'll need to `fmap` an `unignore` function over each of the results in the tests.

**Comment by treeowl** [comment]
My reasoning: your current implementation may not use the `Eq` instance, but that's by no means guaranteed. And relying on implementation details in testing makes it really annoying for someone reading the test to know if it's right.

**Comment by oisdk** [comment]
Yup, makes sense!

File: tests/seq-properties.hs [hunk 3]
```diff
@@ -596,7 +603,8 @@ prop_sortOn (Fun _ f) xs =
 
 prop_sortOnStable :: Seq UnstableOrd -> Bool
 prop_sortOnStable xs =
-    toList' (sortOn ordKey xs) ~= listSortOn ordKey (toList xs)
+    (fmap . fmap) unignore (toList' (sortOn ordKey xs)) ~=
+    fmap unignore (listSortOn ordKey (toList xs))
```
Comment (treeowl) [inline tests/seq-properties.hs]: No, this is wrong. You want

```haskell
Fun A UnstableOrd -> Seq A -> Bool
```

This means `UnstableOrd` will need an `Arbitrary` instance, which should be pretty easy to base on the one for pairs.

**Comment by treeowl** [comment]
Thinking more carefully, we probably don't actually need a separate stability test for `sortOn` after all. But having it does very little harm.

**Pull request merged**